src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -206,6 +213,137 @@ def get_role_policy(self, role_name, policy_name: str | None = None, attached_po
                     policy_actions.append(AWSPolicyAction("s3", privilege, f"s3a://{match.group(1)}"))
         return policy_actions
 
+    def add_uc_role(self, role_name, policy_name, s3_prefixes: set[str], account_id: str, kms_key=None):
```

Comment:
This method is too long and unmaintainable. Refactor every policy document dictionary into own methods to make this method readable

Reply:
OK